frontend/src/App.tsx

```diff
@@ -1,41 +1,24 @@
 import "./App.css";
 import React, { useEffect, useState } from "react";
 import { ProductList } from "./components/ProductsList.component";
-import data from "./db/db.json";
 import { Product } from "./model/model";
 
 const App: React.FC = () => {
-    useEffect(() => {
-        fetch("https://jsonplaceholder.typicode.com/posts", {
-            method: "POST",
-            body: JSON.stringify(data),
-            headers: {
-                "Content-type": "application/json; charset=UTF-8",
-            },
-        })
-            .then((response) => {
-                if (!response.ok) {
-                    throw new Error("Failed to post data");
-                }
-                return response.json();
-            })
-            .catch((error) => console.error("Error posting data:", error));
-    }, []);
-
     const [products, setProducts] = useState<Product[]>([]);
+    console.log("PORT:", import.meta.env.VITE_PORT);
 
     useEffect(() => {
-        fetch("http://localhost:3000/products")
-            .then((response) => {
-                if (!response.ok) {
-                    throw new Error("Failed to fetch data");
-                }
-                return response.json();
-            })
-            .then((json) => {
-                setProducts(json);
-            })
-            .catch((error) => console.error("Error fetching data:", error));
+        async function fetchData() {
+            const response = await fetch(
+                `http://localhost:${import.meta.env.VITE_PORT}/api/products`
```

Comment:
The project was created with Vite (instead of the old react-create-app) and this is the way to support the use of environmental values. I spent quite some time fixing the types as the docs are not great on this subject.